googlemock/include/gmock/internal/gmock-pp.h

```diff
@@ -313,5 +324,65 @@ static_assert(
   GMOCK_PP_INTERNAL_CALL_MACRO(_Macro, _i, _Data, GMOCK_PP_HEAD _Tuple) \
   GMOCK_PP_INTERNAL_FOR_EACH_IMPL_14(GMOCK_PP_INC(_i), _Macro, _Data,   \
                                      (GMOCK_PP_TAIL _Tuple))
+#define GMOCK_PP_INTERNAL_FOR_EACH_IN_CLASS_IMPL_0(_i, _Macro, _Data, _Tuple) \
+  static_assert(true, "")
```

Comment:
Why add this definition?

Reply:
That’s explained in [this comment](https://github.com/google/googletest/pull/2139/commits/bd1773ee7577b10a601b07f59625df36bd339e5e#diff-98584e9a9fab1f527c40de2900753506R163).

The `static_assert` is used to consume the semicolon properly. It’s conceptually similar to the `do {…} while (false)` macro construct, except intended to work in struct/class context when there’s nothing else to go in its place. `static_assert(true, "")` is simply a helpful no-op that consumes a semicolon.

@gennadiycivil, what do you think about this (bd1773ee7577) or its precursor (aef38af5348f)? If it’s easier, I can also propose it for review internally.

Reply:
@markmentovai I like the "easy" ([aef38af](https://github.com/google/googletest/commit/aef38af5348f344d0778d233f9e36699265fdbce))  way but not sure if it would break existing tests. It is easier to review internally , lets go that route for now.

Reply:
> That’s explained in [this comment](https://github.com/google/googletest/pull/2139/commits/bd1773ee7577b10a601b07f59625df36bd339e5e#diff-98584e9a9fab1f527c40de2900753506R163).
> 
> The `static_assert` is used to consume the semicolon properly. It’s conceptually similar to the `do {…} while (false)` macro construct, except intended to work in struct/class context when there’s nothing else to go in its place. `static_assert(true, "")` is simply a helpful no-op that consumes a semicolon.
> 
> @gennadiycivil, what do you think about this ([bd1773e](https://github.com/google/googletest/commit/bd1773ee7577b10a601b07f59625df36bd339e5e)) or its precursor ([aef38af](https://github.com/google/googletest/commit/aef38af5348f344d0778d233f9e36699265fdbce))? If it’s easier, I can also propose it for review internally.

@markmentovai: Got it. Thanks for the description :).
@gennadiycivil: the catch with the easier way is that it's inconsistent with the convention of the other macro definitions; all the other GTEST macros require trailing semicolons, and many other macros I've dealt with in the open source community require trailing semicolons. Departing from this common convention results in a design wart with gmock, which will confuse implementors when they approach this API for the first time.